packages/types/src/interfaces/contracts/types.ts

```diff
@@ -436,6 +439,15 @@ export interface ScheduleTo264 extends Struct {
 /** @name SeedOf */
 export interface SeedOf extends Hash {}
 
+/** @name StorageDeposit */
+export interface StorageDeposit extends Enum {
+  readonly isRefund: boolean;
+  readonly asRefund: Balance;
+  readonly isCharge: boolean;
+  readonly asCharge: Balance;
+  readonly type: 'Refund' | 'Charge';
```

Comment:
Hey, I've a question here: Perhaps it is because of compatibility with legacy code. But why do we need Booleans `isRefund` and `asRefund` when we have `type` which can be either `'Refund'` or `'Charge'`? And why do we need two different Balances? From my intuition, I'd say wen only need the two fields `type: 'Refund' | 'Charge'` and `balance: Balance`?

Reply:
these types are auto generated from substrate metadata